soco/events.py

```diff
@@ -220,23 +220,28 @@ def do_NOTIFY(self):  # pylint: disable=invalid-name
         # find the relevant service from the sid
         with _sid_to_service_lock:
             service = _sid_to_service.get(sid)
-        log.info(
-            "Event %s received for %s service on thread %s at %s", seq,
-            service.service_id, threading.current_thread(), timestamp)
-        log.debug("Event content: %s", content)
-        variables = parse_event_xml(content)
-        # Build the Event object
-        event = Event(sid, seq, service, timestamp, variables)
-        # pass the event details on to the service so it can update its cache.
-        if service is not None:  # It might have been removed by another thread
-            # pylint: disable=protected-access
-            service._update_cache_on_event(event)
-        # Find the right queue, and put the event on it
-        with _sid_to_event_queue_lock:
-            try:
-                _sid_to_event_queue[sid].put(event)
-            except KeyError:  # The key have been deleted in another thread
-                pass
+        if service:
+            log.info(
+                "Event %s received for %s service on thread %s at %s", seq,
+                service.service_id, threading.current_thread(), timestamp)
+            log.debug("Event content: %s", content)
+            variables = parse_event_xml(content)
+            # Build the Event object
+            event = Event(sid, seq, service, timestamp, variables)
+            # pass the event details on to the service so it can update its
+            # cache.
+            if service is not None:
```

Comment:
I guess, strictly speaking, the if check on line 233 is no longer needed, since by your addition it is checked on line 223.